wolfcrypt/src/aes.c

```diff
@@ -9311,8 +9315,91 @@ int wc_AesGcmInit(Aes* aes, const byte* key, word32 len, const byte* iv,
     if (ret == 0) {
         /* Setup with IV if needed. */
         if (iv != NULL) {
-            /* Cache the IV in AES GCM object. */
-            XMEMCPY((byte*)aes->reg, iv, ivSz);
+            /* Initialize and generate GHASH from provided IV. */
+            ret = AesGcmInit_C(aes, iv, ivSz);
```

Comment:
This solution, which uses a call to the software `AesGvmInit()` to condense the IV works with the tests.  It has an awkward `needInit` flag used to handle the case where the IV is NOT set by this function, but by another API, and the cached version in the `aes` structure is used...